### PR TITLE
Update Map selection

### PR DESCRIPTION
**Pull request by JesJehle**
I introduced some changes to the Map. I separated the data selection and the base map selection into different components. 
I added a button to the map which triggers a sheet modal on mobile and a popup on desktop.
What do you think?

### CODE REVIEW

File: src/pages/MapPage.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.mapButton {
+  position: absolute;
+  right: 5%;
+  top: 10%;
```
Comment (mmaelicke) [inline src/pages/MapPage.css:4]: Working with % looks wild and weird on large screens. I would avoid that.
I quite liked the default positioning of the Fabs, that Ionic shipped with.

File: src/pages/MapPage.tsx [hunk 2]
```diff
@@ -23,7 +43,13 @@ const MapPage: React.FC = () => {
       </IonHeader>
 
       <IonContent fullscreen>
+        <MapButton />
         <MainMap />
+        {window.innerWidth > 768 ? (
+          <BaseLayerPopover />
+        ) : (
+          <BaseLayerSheetModal modal={modal}></BaseLayerSheetModal>
+        )}
```
Comment (mmaelicke) [inline src/pages/MapPage.tsx:52]: This does not work. On a Linux machine this is always using the Popover. 
Secondly, the render will not be updated when the innerWidth changes. You need to use useEffect, useCallback or a custom hook for that. 
I suggest that you merge, but dirctly open a bug-issue and assign to me. We need an application-wide `useViewport` hook that gives you the width and triggers if the width changes. We need to also integrate Ionic platform information into it, to better distinguish between screen-size and device information and adjust the UI accordingly.

**Review by mmaelicke — APPROVED** [review]
I will approve as my two comments are minor / hard to implement here, respectively

File: src/pages/MapPage.tsx (hunk 2 above)
Comment (JesJehle) [reply src/pages/MapPage.tsx:52]: It's true that it only works after reloading. But this should be the base case anyway. Normally, the screen size should not change. 
Does this not work on Linux machines?
An application wide hook seems helpful also for further styling.
So, I will merge and open an issue.

**Pull request merged**